ci-operator/step-registry/vsphere/provision/bastionhost/vsphere-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,151 @@
+
+#Get available ova template
+echo "$(date -u --rfc-3339=seconds) - Get avaiable ova template..."
+if openshift-install coreos print-stream-json 2>/tmp/err.txt >/tmp/coreos.json; then
+   echo "Using stream metadata"
+   ova_url=$(jq -r '.architectures.x86_64.artifacts.vmware.formats.ova.disk.location' < /tmp/coreos.json)
+else
+   echo "Unhandled error from openshift-install" 1>&2
+   cat /tmp/err.txt
+   exit 1
+fi
+rm -rf /tmp/err.txt /tmp/coreos.json
+
+vm_template="${ova_url##*/}"
+
+cat > /tmp/rhcos.json << EOF
+{
+   "DiskProvisioning": "thin",
+   "MarkAsTemplate": false,
+   "PowerOn": false,
+   "InjectOvfEnv": false,
+   "WaitForIP": false,
+   "Name": "${vm_template}-bastion",
+   "NetworkMapping":[{"Name":"VM Network","Network":"${BASTION_HOST_SUBNET}"}]
+}
+EOF
+
+if [[ "$(govc vm.info ${vm_template} | wc -c)" -eq 0 ]] && [[ "$(govc vm.info ${vm_template}-bastion | wc -c)" -eq 0 ]]
+then
+    echo "${vm_template} does not exist, creating it from ${ova_url}..."
+    curl -L -o /tmp/rhcos.ova "${ova_url}"
+    govc import.ova -options=/tmp/rhcos.json /tmp/rhcos.ova &
+    vm_template="${vm_template}-bastion"
+    wait "$!"
+fi
+
+echo "ova template: ${vm_template}"
+
+#Create bastion host virtual machine
+echo "$(date -u --rfc-3339=seconds) - Creating bastion host..."
+vm_folder="/${GOVC_DATACENTER}/vm"
+govc vm.clone -vm ${vm_folder}/${vm_template} -on=false -net=${BASTION_HOST_SUBNET} ${bastion_name}
+#govc vm.customize -vm ${vm_folder}/${bastion_name} -name=${bastion_name} -ip=dhcp
+govc vm.change -vm ${vm_folder}/vm/${bastion_name} -c "4" -m "8192" -e disk.enableUUID=TRUE
+disk_name=$(govc device.info -json -vm ${vm_folder}/${bastion_name} | jq -r '.Devices[]|select(.Type == "VirtualDisk")|.Name')
+govc vm.disk.change -vm ${vm_folder}/${bastion_name} -disk.name ${disk_name} -size 100G
+govc vm.change -vm ${vm_folder}/${bastion_name} -e "guestinfo.ignition.config.data.encoding=base64"
+govc vm.change -vm ${vm_folder}/${bastion_name} -e "guestinfo.ignition.config.data=${bastion_ignition_base64}"
+govc vm.power -on ${vm_folder}/${bastion_name}
+
+loop=10
+while [ ${loop} -gt 0 ]; do
+  bastion_ip=$(govc vm.info -json ${vm_folder}/${bastion_name} | jq -r .VirtualMachines[].Summary.Guest.IpAddress)
+  if [ "x${bastion_ip}" == "x" ]; then
+    loop=$(( loop - 1 ))
+    sleep 30
+  else
+    break
+  fi
+done
+
+if [ "x${bastion_ip}" == "x" ]; then
+  echo "Unabel to get ip of bastion host instance ${bastion_name}!"
+  exit 1
+fi
+
+echo "${vm_folder}/${bastion_name}" > "${SHARED_DIR}/bastion_host_path"
+echo "Bastion host created."
+
+#Create dns for bastion host
+echo "$(date -u --rfc-3339=seconds) - Creating bastion host DNS..."
+if [[ -f "${SHARED_DIR}"/basedomain.txt ]]; then
+  base_domain=$(<"${SHARED_DIR}"/basedomain.txt)
+else
+  echo "Unable to find basedomain.txt in SHARED_DIR, set default value..."
+  base_domain="vmc-ci.devcluster.openshift.com"
+fi
+export AWS_DEFAULT_REGION=us-west-2  # TODO: Derive this?
```

Comment:
If not defined this var, then what would happen. In this provision step, it is set to `us-west-2`, while in deprovision step, it is `us-east-1`.

Reply:
let me remove it to see the results.

Reply:
it works after removing that env from both provision step and deprovision step,